sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -531,20 +533,30 @@ object GpuToTimestamp extends Arm {
     FIX_SINGLE_DIGIT_SECOND
   )
 
-  def daysScalarSeconds(name: String): Scalar = {
-    Scalar.timestampFromLong(DType.TIMESTAMP_SECONDS, DateUtils.specialDatesSeconds(name))
-  }
-
-  def daysScalarMicros(name: String): Scalar = {
-    Scalar.timestampFromLong(DType.TIMESTAMP_MICROSECONDS, DateUtils.specialDatesMicros(name))
-  }
-
   def daysEqual(col: ColumnVector, name: String): ColumnVector = {
     withResource(Scalar.fromString(name)) { scalarName =>
       col.equalTo(scalarName)
     }
   }
 
+  /**
+   * Replace special date strings such as "now" with timestampDays. This method does not
+   * close the `stringVector` and `specialValues`.
+   */
+  def replaceSpecialDates(
+      stringVector: ColumnVector,
+      chronoVector: ColumnVector,
+      specialNames: Seq[String],
+      specialValues: Seq[Scalar]): ColumnVector = {
+    specialValues.zip(specialNames).foldLeft(chronoVector) { case (buffer, (scalar, name)) =>
```

Comment:
If we make the Map value a function that produces the scalar, then we won't need to pull the map apart just so we can zip it back together again.Just call `.toSeq` on it, so we could pass it in directly here.

Reply:
Good idea! I made the change.